pkg/kn/commands/flags/sink.go

```diff
@@ -49,26 +57,32 @@ func (i *SinkFlags) AddWithFlagName(cmd *cobra.Command, fname, short string) {
 		"'" + flag + " ksvc:mysvc:mynamespace' for a Knative service 'mysvc' in another namespace 'mynamespace', " +
 		"'" + flag + " https://event.receiver.uri' for an URI with an 'http://' or 'https://' schema, " +
 		"'" + flag + " ksvc:receiver' or simply '" + flag + " receiver' for a Knative service 'receiver' in the current namespace. " +
+		"'" + flag + " special.eventing.dev/v1alpha1/channels:pipe' for GroupVersionResource of v1alpha1 'pipe'. " +
+		"GroupVersionResource requires resource name in a lower case plural form. The inputs are sanitized on best effort basis, " +
```

Comment:
"requires resource name in a lower case plural form" --> Can't we do at least the lower casing on our own so that that case doesn't mind ? Also the plural form is not intuitive if you don't know about resources. I would accept also both, naively assuming that plural form mean adding a `s` (or is there an easier way to find out ?)

The overall help message now became quite a bit lengthy and since it is used for every sink type and some commands have multiple sink options (eg `--sink`, `--sink-reply`, `--sink-dead-letter`), the help page becomes too long.

We might:

* Shorten it and reference to website for the full documentation.
* Try to keep it but only keep the absolutely necessary information.

Eg. instead of 

```
-s, --sink string               Addressable sink for events. You can specify a broker, channel, Knative service or URI. Examples: '--sink broker:nest' for a broker 'nest', '--sink channel:pipe' for a channel 'pipe', '--sink ksvc:mysvc:mynamespace' for a Knative service 'mysvc' in another namespace 'mynamespace', '--sink https://event.receiver.uri' for an URI with an 'http://' or 'https://' schema, '--sink ksvc:receiver' or simply '--sink receiver' for a Knative service 'receiver' in the current namespace. '--sink special.eventing.dev/v1alpha1/channels:pipe' for GroupVersionResource of v1alpha1 'pipe'. GroupVersionResource requires resource name in a lower case plural form. The inputs are sanitized on best effort basis, but in case of 'resource not found' error it may help to double check and correct the input. If a prefix is not provided, it is considered as a Knative service in the current namespace. If referring to a Knative service in another namespace, 'ksvc:name:namespace' combination must be provided explicitly.
```

we might use something like

```
-s, --sink string               Addressable sink for events. Use one of these formats: broker:<name>, channel:<name>, ksvc:<service>:<namespace>, <URI>, or simply <service>. Examples: '--sink broker:nest', '--sink ksvc:mysvc:mynamespace', '--sink https://event.receiver.uri', '--sink receiver'. If the prefix is omitted, it's assumed to be a Knative service in the current namespace. Also supports GroupVersionResource format like 'special.eventing.dev/v1alpha1/channels:pipe'. On errors like 'resource not found', verify your input format.
```

Reply:
I agree, it was a bit long even without this change. Sure, the code tries to do both lower case of course and add naive `+s`. However, we never explained anything from that. I kind of like the idea of a little how to use sinks page with a couple of examples. 

I can move the plural form recommendation into error message maybe?

Reply:
Maybe we might leave of the error explanation, too ?

Reply:
OK, let me remove the additional lines and keep just GVR example to check how that looks.